pkg/gofr/datasource/scylladb/scylladb_test.go

```diff
@@ -0,0 +1,527 @@
+			mockDeps.mockIter.EXPECT().Scan(gomock.Any()).AnyTimes()
+		}, &mockStruct, nil},
+		{"failure case: dest is not pointer", mockStructSlice, func() {
+			mockDeps.mockLogger.EXPECT().Debug(gomock.AssignableToTypeOf(&QueryLog{}))
+		}, mockStructSlice,
+			nil},
+		{"failure case: dest is int", &mockInt, func() {
+			mockDeps.mockLogger.EXPECT().Debug(gomock.AssignableToTypeOf(&QueryLog{}))
+			mockDeps.mockSession.EXPECT().Query(query).Return(mockDeps.mockQuery).AnyTimes()
+			mockDeps.mockQuery.EXPECT().Iter().Return(mockDeps.mockIter).AnyTimes()
+		}, &mockInt, nil},
+		{"failure case: dest is int", &mockInt, func() {
+			mockDeps.mockLogger.EXPECT().Debug(gomock.AssignableToTypeOf(&QueryLog{}))
+			mockDeps.mockSession.EXPECT().Query(query).Return(mockDeps.mockQuery).Times(1)
+			mockDeps.mockQuery.EXPECT().Iter().Return(mockDeps.mockIter).Times(1)
+		}, &mockInt, nil},
+	}
+
+	for i, tc := range testCases {
+		tc.mockCall()
+
+		err := client.Query(&mockStructSlice, query)
+
+		assert.Equalf(t, tc.expRes, tc.dest, "TEST[%d], Failed.\n%s", i, tc.desc)
+		assert.Equalf(t, tc.expErr, err, "TEST[%d], Failed.\n%s", i, tc.desc)
+	}
+}
+
+func Test_Exec(t *testing.T) {
+	const query = "INSERT INTO Users (id, name) VALUES(1, 'Test')"
+
+	client, mockDeps := initTest(t)
+
+	testCases := []struct {
+		desc     string
+		mockCall func()
+		expErr   error
+	}{
+		{"success case", func() {
+			mockDeps.mockLogger.EXPECT().Debug(gomock.AssignableToTypeOf(&QueryLog{}))
+			mockDeps.mockSession.EXPECT().Query(query, nil).Return(mockDeps.mockQuery).AnyTimes()
+			mockDeps.mockQuery.EXPECT().Exec().Return(nil).Times(1)
+		}, nil},
+		{"failure case", func() {
+			mockDeps.mockLogger.EXPECT().Debug(gomock.AssignableToTypeOf(&QueryLog{}))
+			mockDeps.mockSession.EXPECT().Query(query, nil).Return(mockDeps.mockQuery).AnyTimes()
+			mockDeps.mockQuery.EXPECT().Exec().Return(errMock).Times(1)
+		}, errMock},
+	}
+
+	for i, tc := range testCases {
+		tc.mockCall()
+
+		err := client.Exec(query)
+
+		assert.Equalf(t, tc.expErr, err, "TEST[%d], Failed.\n%s", i, tc.desc)
+	}
+}
+
+func Test_NewBatch(t *testing.T) {
+	const batchName = "testBatch"
+
+	client, mockDeps := initTest(t)
+
+	testCases := []struct {
+		desc      string
+		batchType int
+		mockCall  func()
+		expErr    error
+	}{
+		{"valid log type", LoggedBatch, func() {
+			mockDeps.mockSession.EXPECT().newBatch(gocql.BatchType(LoggedBatch)).Return(&scyllaBatch{}).Times(1)
+		}, nil},
+		{"valid log type, empty batches", LoggedBatch, func() {
+			client.scylla.batches = nil
+
+			mockDeps.mockSession.EXPECT().newBatch(gocql.BatchType(LoggedBatch)).Return(&scyllaBatch{}).Times(1)
+		}, nil},
+		{"invalid log type", -1, func() {}, errUnsupportedBatchType},
+	}
+
+	for i, tc := range testCases {
+		tc.mockCall()
+
+		err := client.NewBatch(batchName, tc.batchType)
+
+		assert.Equalf(t, tc.expErr, err, "TEST[%d], Failed.\n%s", i, tc.desc)
```

Comment:
Error should be tested with


```suggestion
		assert.Errorf(t, err, tc.expErr, "TEST[%d], Failed.\n%s", i, tc.desc)
```

Reply:
I have some test cases where the error is nil in success cases. However, if I use errorf, I have to provide an error for the success case as well. Additionally, the third argument in errorf expects a string format, and it doesn't accept an error type directly.